test/ranges.jl

```diff
@@ -1326,6 +1326,15 @@ end
     @test r[4] === 3.0+im
 end
 
+@testset "issue #37276" begin
+    r = range(Complex{BigFloat}(0), stop=Complex{BigFloat}(1), length=4)
+    @test isa(@inferred(r[2]), Complex{BigFloat})
+    @test r[1] == 0.
+    @test r[2] ≈ big"1"/3
```

Comment:
Is this perhaps a bit too loose? Would it make sense to set a lower tolerance here?

Reply:
Are you referring to the tolerance of `isapprox`?
AFAIU, the default tolerance used by `isapprox` is given by the square root of `eps` of the types of the operands, which we know to be `BigFloat`s here. In this case, this means a relative tolerance of approximately 4e-39 (which makes everything fail if any part of the computation is performed in double precision).

But you're right that it is much higher than the few ulps that we could expect; we could very well set this to something like 1e-75 and still be safe. TBH, the reason why I kept the default tolerance parameters is because this seems to be how it's done for all other approximate equality tests in `test/ranges.jl`.

TL;DR: I don't have any strong opinion about this. I can very well change this tolerance to any value deemed more appropriate.

Reply:
Yes, that's what I meant. I think it's better to be more conservative with tests like these, because it's always better to discover regressions early. Perhaps the other tests should be more strict as well, but that's probably a separate discussion.

Reply:
OK. I agree that more strictness is always a good thing in tests. I amended to commit to include a relative tolerance of `10eps(BigFloat)`, which I guess makes the intent explicit (although in this particular case results are exactly rounded, I think it is more future-proof to not rely on this and allow an error of a few ulps, which should correspond to the expectation users should have in the general case)